packages/styles/schematics/migrations/general/classes/rtl.ts

```diff
@@ -4,90 +4,114 @@ import { breakpoints, sizes } from '../../../utils/constants';
+    const selectorProperties = ['m', 'p'];
+    const selectorSides = ['l', 'r'];
+    const selectorBreakpoints = [''].concat(breakpoints);
+
+    return selectorProperties
+      .map(property =>
+        selectorSides.map(side =>
+          selectorBreakpoints.map(breakpoint =>
+            sizes.map(size => `.${property}${side}${classPart(breakpoint)}${classPart(size)}`),
+          ),
+        ),
+      )
+      .join(', ');
   }
 }
 
 class AlignmentClassesUpdate implements DomUpdate {
   cssClassRegex = new RegExp(`^(float|text)(?:-(${breakpoints.join('|')}))?-(left|right)$`);
-  sideUpdate = new Map([ [ 'left', 'start' ], [ 'right', 'end' ] ]);
+  sideUpdate = new Map([
+    ['left', 'start'],
+    ['right', 'end'],
+  ]);
 
   selector = this.createSelectors();
 
   update($elements: Cheerio<AnyNode>, $: CheerioAPI) {
-    $elements
-      .each((_i, element) => {
-        const $element = $(element);
-
-        $element
-          .attr('class')
-          ?.split(' ')
-          .forEach(cssClass => {
-            const match = cssClass.match(this.cssClassRegex);
-
-            if (match) {
-              const property = match[1];
-              const breakpoint = match[2];
-              const side = match[3];
-
-              $element
-                .removeClass(cssClass)
-                .addClass(`${property}${breakpoint ? `-${breakpoint}` : ''}-${this.sideUpdate.get(side)}`);
-            }
-          });
-      });
+    $elements.each((_i, element) => {
+      const $element = $(element);
+
+      $element
+        .attr('class')
+        ?.split(' ')
+        .forEach(cssClass => {
+          const match = cssClass.match(this.cssClassRegex);
+
+          if (match) {
+            const property = match[1];
+            const breakpoint = match[2];
+            const side = match[3];
+
+            $element
+              .removeClass(cssClass)
+              .addClass(
+                `${property}${classPart(breakpoint)}${classPart(this.sideUpdate.get(side))}`,
+              );
+          }
+        });
+    });
   }
 
   createSelectors() {
-    const selectorProperties = [ 'float', 'text' ];
-    const selectorBreakpoints = [ '' ].concat(breakpoints);
-    const selectorSides = [ 'left', 'right' ];
+    const selectorProperties = ['float', 'text'];
+    const selectorBreakpoints = [''].concat(breakpoints);
+    const selectorSides = ['left', 'right'];
+
+    return selectorProperties
+      .map(property =>
+        selectorBreakpoints.map(breakpoint =>
+          selectorSides.map(side => `.${property}${classPart(breakpoint)}${classPart(side)}`),
+        ),
+      )
+      .join(', ');
+  }
+}
 
-    return selectorProperties.map(property => selectorBreakpoints.map(breakpoint => selectorSides.map(side => `.${property}${breakpoint
-      ? `-${breakpoint}`
-      : ''}-${side}`))).join(', ');
+function classPart(partName: string | undefined) {
+  if (!partName) {
+    return '';
   }
+
+  return `-${partName}`;
```

Comment:
The real addition is this function + the usage of it. I don't know how to test it though.

Reply:
It seems ok to me!
Maybe just rename `classPart` to `optionalClassPart` to make it more meaningful.